tables/cost_and_usage_focus/cost_and_usage_focus_table.go

```diff
@@ -0,0 +1,90 @@
+package cost_and_usage_focus
+
+import (
+	"strings"
+	"time"
+
+	"github.com/rs/xid"
+	typehelpers "github.com/turbot/go-kit/types"
+	"github.com/turbot/pipe-fittings/v2/utils"
+	"github.com/turbot/tailpipe-plugin-aws/sources/s3_bucket"
+	"github.com/turbot/tailpipe-plugin-sdk/artifact_source"
+	"github.com/turbot/tailpipe-plugin-sdk/artifact_source_config"
+	"github.com/turbot/tailpipe-plugin-sdk/constants"
+	"github.com/turbot/tailpipe-plugin-sdk/row_source"
+	"github.com/turbot/tailpipe-plugin-sdk/schema"
+	"github.com/turbot/tailpipe-plugin-sdk/table"
+)
+
+const CostUsageFocusTableIdentifier = "aws_cost_and_usage_focus"
+
+type CostUsageFocusTable struct{}
+
+func (c *CostUsageFocusTable) Identifier() string {
+	return CostUsageFocusTableIdentifier
+}
+
+func (c *CostUsageFocusTable) GetSourceMetadata() []*table.SourceMetadata[*CostUsageFocus] {
+	defaultS3ArtifactConfig := &artifact_source_config.ArtifactSourceConfigImpl{
+		FileLayout: utils.ToStringPointer("%{DATA:export_name}/(?:data/%{DATA:partition}/)?(?:%{INT:from_date}-%{INT:to_date}/)?(?:%{DATA:assembly_id}/)?(?:%{DATA:timestamp}-%{DATA:execution_id}/)?%{DATA:file_name}.csv.(?:zip|gz)"),
```

Comment:
Can the FOCUS reports have `%{INT:from_date}-%{INT:to_date}`, or is that pattern mostly found in legacy reports? Similarly, can they have `%{DATA:assembly_id}` and end in `.zip`?

For any file created by AWS Data Export, I thought it'd have to be one of these patterns:
- Create new files: `<bucket-name>/<prefix>/<export-name>/data/<partition>/<timestamp>-<execution-id>/<export-name>-<chunk-number>.csv.gz`
- Overwrite: `<bucket-name>/<prefix>/<export-name>/data/<partition>/<export-name>-<chunk-number>.csv.gz`

Reply:
Seems like the pattern could be simpler if we only accounted for the 2 cases above (unless there are other possible ones not listed in https://docs.aws.amazon.com/cur/latest/userguide/dataexports-export-delivery.html#export-summary)

Reply:
Refined the pattern specifically for FOCUS 1.0. It now exclusively matches files following this structure:  `%{DATA:export_name}/data/%{DATA:partition}/(?:%{TIMESTAMP_ISO8601:timestamp}-%{UUID:execution_id}/)?%{DATA:filename}.csv.gz`